Simplenote/src/main/java/com/automattic/simplenote/NoteListFragment.java

```diff
@@ -447,6 +457,19 @@ public void onResume() {
         super.onResume();
         getPrefs();
         refreshList();
+        mBucketPreferences.start();
+        mBucketPreferences.addOnDeleteObjectListener(this);
+        mBucketPreferences.addOnNetworkChangeListener(this);
+        mBucketPreferences.addOnSaveObjectListener(this);
```

Comment:
What's the purpose of putting these all in `NoteListFragment` instead of in their own class, something like a `PreferencesManager` and connecting changes/updates to that class vs. connecting to a smaller API like `mPreferencesManager.addChangeListener(this)`

Just asking out of curiosity - trying to learn the patterns and how we make decisions on Android

Reply:
Sure, but isn't the _only_ thing that we're doing is updating the recent search list?

In other words, for the sake of the UI and the application, I'm curious why we care about these internal details in the `NoteListFragment`.

Is there anything done here that would be specific to one of those three things that couldn't be done automatically inside some kind of `RecentSearchManager`?

I'm asking because this module is already pretty big and the addition of this stuff and its new implementation functions is even more confusing when I'm trying to understand how it works and how the pieces fit together 😀

```java
RecentSearchManager recentSearches = getRecentSearchManager();

recentSearches.addOnChangeListener(new Consumer<ArrayList<String>>() {
	@Override
	public void accept(ArrayList<String> searches) {
		diffAndUpdateSearchAdapter(…);
	}
} );

new View.OnClickListener() {
	@Override
	public void onClick(View view) {
		recentSearches.add(item);
	}
}

holder.mButtonDelete.setOnClickListener(new View.OnClickListener() {
    @Override
    public void onClick(View view) {
        if (!isAdded()) {
            return;
        }

        final String item = holder.mSuggestionText.getText().toString();
        recentSearches.drop(item);

        Snackbar
            .make(getRootView(), R.string.snackbar_deleted_recent_search, Snackbar.LENGTH_LONG)
            .setActionTextColor(ThemeUtils.getColorFromAttribute(requireContext(), R.attr.colorAccent))
            .setAction(
                getString(R.string.undo),
                new View.OnClickListener() {
                    @Override
                    public void onClick(View view) {
                        recentSearches.restoreLastSearch();
                    }
                }
            )
            .show();
    }
});
```